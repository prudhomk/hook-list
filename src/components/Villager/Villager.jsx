import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Villager = ({ name, species, phrase, image, _id }) => (
  <Link to={`/${_id}`}>
    <figure>
      <img src={image} alt={name} />
      <figcaption>
        <p>{name}</p>
        <p>{species}</p>
        <p>{phrase}</p>
      </figcaption>
    </figure>
  </Link>
);

Villager.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  phrase: PropTypes.string.isRequired,
  image: PropTypes.string.isReqruied,
};

export default Villager;