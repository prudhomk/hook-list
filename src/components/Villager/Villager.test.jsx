import React from 'react';
import { render } from '@testing-library/react';
import Villager from './Villager';
import { MemoryRouter } from 'react-router-dom'

describe('Villager component', () => {
  it('renders Villager', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Villager
          name="Booger"
          species="Snot"
          phrase="Ooga Booga"
          image="https://image.com/image.jpeg"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
