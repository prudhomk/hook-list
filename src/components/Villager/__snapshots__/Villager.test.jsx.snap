// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Villager component renders Villager 1`] = `
<DocumentFragment>
  <a
    href="/undefined"
  >
    <figure>
      <img
        alt="Booger"
        src="https://image.com/image.jpeg"
      />
      <figcaption>
        <p>
          Booger
        </p>
        <p>
          Snot
        </p>
        <p>
          Ooga Booga
        </p>
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
